src/rollup/display-conditions/scroll/register-position-watcher.js

```diff
@@ -15,14 +15,19 @@ import document from '../../dom/document';
 export default function registerPositionWatcher (percent) {
   var watcher = {
     check: function () {
-      var height = Math.max(document.body.scrollHeight, document.body.offsetHeight, document.documentElement.clientHeight, document.documentElement.scrollHeight, document.documentElement.offsetHeight),
-          positionInPixels = height * (percent / 100),
-          offset = document.documentElement.scrollTop || document.body.scrollTop;
+      var height = Math.max(
+            document.body.scrollHeight,
+            document.body.offsetHeight,
+            document.documentElement.clientHeight,
+            document.documentElement.scrollHeight,
+            document.documentElement.offsetHeight
+          ),
+          offset = document.documentElement.scrollTop || document.body.scrollTop,
+          clientHeight =
+          document.documentElement.clientHeight || document.body.clientHeight,
+          percentageScrolled = (offset / (height - clientHeight)) * 100;
```

Comment:
Noted that `document.documentElement.clientHeight` is used in both `height` and clientHeight` is there a potential that this divisor will be zero at some point resulting in a NaN? Maybe we need to remove `clientHeight` from the max calculation for `height`? :thinking:

Reply:
I refactored the arithmetic slightly but you're right - it is possible to end up with a `NaN` dividend in a scenario where `scrollHeight` is equal to `clientHeight`.

However, that scenario means there isn't any scrolling possible anyway, so our function should return `false`, which it does, as anything compared to `NaN` will do:

```js
  // if NaN, will always return `false`
  return percentageScrolled >= percent;
```

So, I _think_ that should be sufficient unless you'd like to be more explicit in handling `NaN` but maybe the comment is enough context?